integration-tests/pkg/mock_sensor/expect_conn.go

```diff
@@ -72,14 +72,47 @@ loop:
 			if conn.GetContainerId() != containerID {
 				continue loop
 			}
-
 			if len(s.Connections(containerID)) == n {
 				return s.Connections(containerID)
 			}
 		}
 	}
 }
 
+// ExpectSameElementsConnections compares a list of expected connections to the observed connections. This comparison is done at the beginning, when a new
+// connection arrives, and after a timeout period. The number of connections must match and the expected and observed connections must match, but the order
+// does not matter.
+func (s *MockSensor) ExpectSameElementsConnections(t *testing.T, containerID string, timeout time.Duration, expected ...types.NetworkInfo) bool {
+	types.SortConnections(expected)
+
+	equal := func(c1 types.NetworkInfo, c2 types.NetworkInfo) bool {
```

Comment:
[nit] This can be made a bit less verbose
```suggestion
	equal := func(c1, c2 types.NetworkInfo) bool {
```